htdocs/core/lib/website2.lib.php

```diff
@@ -104,10 +104,12 @@ function dolSavePageAlias($filealias, $object, $objectpage)
 	}
 	// Save also alias into all language subdirectories if it is a main language
 	elseif (empty($objectpage->lang) || !in_array($objectpage->lang, explode(',', $object->otherlang))) {
-		if (empty($conf->global->WEBSITE_DISABLE_MAIN_LANGUAGE_INTO_LANGSUBDIR) && !empty($object->otherlang)) {
+		if (empty($conf->global->WEBSITE_DISABLE_MAIN_LANGUAGE_INTO_LANGSUBDIR)) {
```

Comment:
It is strange to remove the test on "&& !empty($object->otherlang)"
If $object->otherlang is empty, then we won't execute the loop late based on $object->otherlang (it means we have nothing to do)
I think you can keep this test.

Reply:
I removed it because finally test if `$object->otherlang` is empty is not a reliable mean to avoid to have an empty `$subdir` in the loop bellow. We can have `$object->otherlang` with value like this : `es,,en` or `es,en,` (may be an user input error), if is the case the array returned by `explode(',', $object->otherlang)` will result get an element with an empty string which will result to get a `$subdir` with empty string in the for loop even if `$object->otherlang` is not empty. I figured that testing if `$object->otherlang` is empty will give the a false security idea so i removed it. If you think is better to keep it i will re-add it.